specification/dartLangSpec.tex

```diff
@@ -19655,42 +19560,153 @@ \subsection{Type Aliases}
-\code{\TYPEDEF{} $T$ \id($T_1\ p_1, \ldots,\ T_n\ p_n,\ \{T_{n+1}\ p_{n+1}, \ldots,\ T_{n+k}\ p_{n+k}\}$);}
+is treated as
 
 \noindent
-declared in a library $L$ is to introduce the name \id{} into the scope of $L$, bound to the function type
-\FunctionTypeSimple{T}{\List{T}{1}{n},\ \{\PairList{T}{p}{n+1}{n+k}\}}.
+\code{\TYPEDEF{} \id<\TypeParametersStd> =}
+
+\noindent
+\code{\quad$S?$ \FUNCTION($T_1\ p_1, \ldots,\ T_n\ p_n,\ \{T_{n+1}\ p_{n+1}, \ldots,\ T_{n+k}\ p_{n+k}\}$);}
+
+\LMHash{}%
+In these rules, for each $j$, if $T_j?$ is empty
+then \DefineSymbol{T_j} is \DYNAMIC,
+otherwise $T_j$ is $T_j?$.
+
+\commentary{%
+This means that the older forms allow for a parameter type to be omitted,
+in which case it is taken to be \DYNAMIC,
+but the parameter name cannot be omitted.
+This behavior is error prone,
+and hence the newer form (with \lit{=}) is recommended.
+For instance, the declaration \code{\TYPEDEF{} $F$(int);}
+specifies that $F$ denotes the type
+\code{\DYNAMIC{} \FUNCTION(\DYNAMIC)},
+and it is documented, but technically ignored,
+that the parameter has the name \code{int}.
+It is extremely likely that a reader will misread this,
+and assume that \code{int} is the parameter type.%
+}
+
+\LMHash{}%
+It is a compile-time error if a default value is specified for
+a formal parameter in these older forms,
+or if a formal parameter has the modifier \COVARIANT.
+
+\commentary{%
+Note that the old forms can only define function types,
+and they cannot denote the type of a generic function.
+When such a type alias has type parameters,
+it always expresses a family of non-generic function types.
+These restrictions exist because that syntax was defined
+before generic functions were added to Dart.%
+}
 
 \LMHash{}%
-In either case, if{}f no return type is specified, it is taken to be \DYNAMIC{}.
-Likewise, if a type annotation is omitted on a formal parameter, it is taken to be \DYNAMIC{}.
+Let $D_F$ be a type alias declaration of the form
+
+\noindent
+\code{\TYPEDEF{} $F$<\TypeParametersStd> = $T$;}
+
+\noindent
+If $T$ or $B_j$ for some $j \in 1 .. s$ is or contains
+a \synt{typeName} $G$ denoting a type alias declaration $D_G$,
+then we say that
+\IndexCustom{$D_F$ depends on $D_G$}{type alias!dependency}.
+
+%%TODO(eernst): The relaxation where a type alias dependency arises
+%% via $T$, but not via $B_j$, is more permissive and allows for some
+%% type aliases to be declared that will be an error with the above rule.
+%% It is very likely that the more relaxed rule can be used. This is a
+%% possible enhancement for the future, and a non breaking change.
 
 \LMHash{}%
-It is a compile-time error if any default values are specified
-in the signature of a function type in a type alias.
-%A typedef may only refer to itself via the bounds of its generic parameters.
-Any self reference in a type alias,
-either directly or recursively via another type declaration,
-is a compile-time error.
+Let $D$ be a type alias delaration,
```

Comment:
Spelling:  delaration -> declaration

Reply:
Done in https://github.com/dart-lang/language/pull/1555.